.github/workflows/e2e-pr.yml

```diff
@@ -0,0 +1,99 @@
+name: E2E - PR
+on:
+  pull_request:
+    branches:
+      - main
+
+# Cancel any in progress run of the workflow for a given PR
+# This avoids building outdated code
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  android:
+    name: Android
+    uses: ./.github/workflows/e2e-android.yml
+    with:
+      android-api-level: 30
```

Comment:
We _can_ run it against 31, but for some reason tests against 31 run way slower. Something goes wrong when initializing tests the first time through, and the whole suite fails. Instead of taking ~15 minutes, tests instead take ~35 minutes. I'm hoping to look into this in a follow-up PR.